queue_reconstruction_by_height.java

```diff
@@ -0,0 +1,51 @@
+// Time Complexity : O(n^2), where n is the number of rows in the 2D array
+// Space Complexity : O(n), where n is the number of rows in the 2D array (arraylist space) 
+// Did this code successfully run on Leetcode : Yes
+// Any problem you faced while coding this : I faced to problems
+                                            //1. Not sure if my space complexity is correct, as we are creating an arraylist to store
+                                                    //the result, but we are not returning it as it, we are converting to 2D array first.
+                                                    //so should it be O(n) only or O(1)??
+                                            //2. I am not understanding, if how the conversion of Arraylist of arrays To Array of arrays
```

Comment:
when a creating a 2D array with new int[rowDimension][columnDimension], the columnDimension can be left empty, only the rowDimension is mandatory. The columnDimension can be created when each row is created.

int[][] a = new int[2][];
a[0] = new int[5];
a[1] = new int[10]; 

This is valid. As per the same logic each row in the list will be created after toArray() is invoked.

For the rowDimension you can pass anything like 0 or list.size. It will pick the right value based on the list's size